src/main.rs

```diff
@@ -260,67 +261,26 @@ fn search_config(rule: &Path) -> Option<PathBuf> {
 fn parse_filelist(
     path: &Path,
 ) -> Result<(Vec<PathBuf>, Vec<PathBuf>, HashMap<String, Option<Define>>), Error> {
-    let mut f =
-        File::open(path).with_context(|| format!("failed to open '{}'", path.to_string_lossy()))?;
-    let mut s = String::new();
-    let _ = f.read_to_string(&mut s);
-
-    let mut files = Vec::new();
-    let mut includes = Vec::new();
+    let path_str = path.to_string_lossy();
+    let filelist = match verilog_filelist_parser::parse_file(&path_str) {
```

Comment:
`verilog_filelist_parser::parse_file(path: &str)` may cause trouble on some environments using non-UTF8 file path ( ex. Windows ).
I think the following prototype is better.

```rust
pub fn parse_file<T: AsRef<Path>>(path: T) -> Result<Filelist, Box<dyn Error>> {
    let path = path.as_ref();
}
```

By using `AsRef<Path>`, `parse_file` can take `Path`, `PathBuf`, `String`, `str`..
It is convenient.